tests/xharness/Xharness.Tests/Tests/AppRunnerTests.cs

```diff
@@ -464,10 +477,20 @@ public async Task RunOnSimulatorSuccessfullyTest ()
 				.ReturnsAsync (new ProcessExecutionResult () { ExitCode = 0 });
 
 			var xmlResultFile = Path.ChangeExtension (testResultFilePath, "xml");
-			var resultParser = new Mock<IResultParser> ();
-			resultParser
-				.Setup (x => x.CleanXml (testResultFilePath, xmlResultFile))
-				.Callback (() => File.Copy (testResultFilePath, xmlResultFile));
+			var testReporterFactory = new Mock<ITestReporterFactory> ();
+			var testReporter = new Mock<ITestReporter> ();
+			testReporterFactory.Setup (f => f.Create (
+				It.IsAny<IAppRunner> (),
+				It.IsAny<string> (),
+				It.IsAny<ISimpleListener> (),
+				It.IsAny<ILog> (),
+				It.IsAny<ICrashSnapshotReporter> ())).Returns (testReporter.Object);
```

Comment:
It would be better if instead of `IsAny` you matched the values it should receive to be sure we for example don't send in wrong deviceName or something.